### PR TITLE
 Implement a SpectrumList loader for specviz

### PR DESCRIPTION
**Pull request by ibusko**
### Description

This pull request enables specviz to load multiple spectrum files from a directory, using the loading abilities of `SpectrumList`:

`specviz.load_spectrum(<path-to-dir>)`

The first spectrum in the list is automatically displayed. Still TBD is a way to combine/overplot the entire set of spectra.

Fixes #<Issue Number>

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [ ] Is a milestone set? Milestone is only currently required for PRs related to Imviz MVP.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/configs/specviz/plugins/parsers.py [hunk 1]
```diff
@@ -50,6 +50,11 @@ def specviz_spectrum1d_parser(app, data, data_label=None, format=None, show_in_v
             except IORegistryError:
                 # Multi-extension files may throw a registry error
                 data = SpectrumList.read(str(path), format=format)
+        elif path.is_dir():
+            data = SpectrumList.read(str(path), format=format)
+            # temporary labelling, just for testing. Better labels should
+            # come from the file list
+            data_label = ["Spectrum " + str(i) for i in range(len(data))]
```
Comment (pllim) [inline jdaviz/configs/specviz/plugins/parsers.py:57]: Nitpick... though a deeper issue is this line ignores any user-provided `data_label`... 🤷 
```suggestion
            data_label = [f"Spectrum {i}" for i in range(len(data))]
```

Comment (ibusko) [reply jdaviz/configs/specviz/plugins/parsers.py:57]: Indeed, you're right. The code is missing a way to allow the user to define the labels. I will fix that...

**Review by pllim — COMMENTED** [review]
If someone provides a spectrum list but scalar data label, how is it supposed to work?

Is it possible to add some tests?

**Comment by rosteen** [comment]
There's already code from my older `SpectrumList` PR that checks to see if the resulting data is a `SpectrumList` and if the data_label isn't a list/tuple, adds numbers to generate the right amount of labels.

**Comment by pllim** [comment]
> There's already code

Then why is codecov complaining? 🤔

**Comment by rosteen** [comment]
Because I was responding to this
> If someone provides a spectrum list but scalar data label, how is it supposed to work?

Not this 
> Is it possible to add some tests?

**Comment by ibusko** [comment]
1 - "If someone provides a spectrum list but scalar data label, how is it supposed to work?": I don't know. The labels list must be paired with the files list. Is adding a running counter to scalar the label OK? Or should we forbid a labels list that is not paired?

2 - "Is it possible to add some tests?": Tests for reading from a list of files or a `SpectrumList` instance are already in place. I tried to add a test for reading from a directory. That is: build a temp directory, copy the test spec1d file (that is passed to the test code) there a few times, each time with a different name, and run the code on that directory. To copy, we need to use `Spectrum1D.write()`, which uses astropy io, which doesn't provide a way to write a x1d file or the like.  

```
    def test_load_spectrum_list_directory(self):
        NSPEC = 3
        with tempfile.TemporaryDirectory() as tmp_dir:
            for i in range(NSPEC):
                file_path = os.path.join(tmp_dir, 'sp'+str(i)+'_x1d.fits')
                with open(file_path, 'wb') as tmp_file:
                    self.spec.write(tmp_file, format='tabular-fits')

            self.spec_app.load_spectrum(tmp_dir)
            assert len(self.spec_app.app.data_collection) == NSPEC
```

**Comment by pllim** [comment]
> Tests for reading from a list of files or a SpectrumList instance are already in place.

Then I don't understand why codecov says it is not covered. I am going to recuse myself from further review. Devs who are familiar with spectrum list handling should review/approve this.

**Comment by ibusko** [comment]
@plim the part that is not covered by tests is the part that reads from a directory. I'm happy to provide a test once I find a way to create physical x1d files in a temp directory.

File: jdaviz/configs/specviz/plugins/parsers.py [hunk 2]
```diff
@@ -40,6 +40,8 @@ def specviz_spectrum1d_parser(app, data, data_label=None, format=None, show_in_v
         data_label = [data_label]
     elif isinstance(data, SpectrumList):
         pass
+    elif isinstance(data, list):
+        data = SpectrumList.read(data, format=format, data_label=data_label)
```
Comment (rosteen) [inline jdaviz/configs/specviz/plugins/parsers.py:44]:
```suggestion
        data = SpectrumList.read(data, format=format)
```

Passing `data_label` into `SpectrumList` doesn't do anything, right? It's not a keyword that's used by specutils, it's handled by the parser code later on.

File: jdaviz/configs/specviz/plugins/parsers.py [hunk 3]
```diff
@@ -49,7 +51,9 @@ def specviz_spectrum1d_parser(app, data, data_label=None, format=None, show_in_v
                 data_label = [data_label]
             except IORegistryError:
                 # Multi-extension files may throw a registry error
-                data = SpectrumList.read(str(path), format=format)
+                data = SpectrumList.read(str(path), format=format, data_label=data_label)
+        elif path.is_dir():
+            data = SpectrumList.read(str(path), format=format, data_label=data_label)
```
Comment (rosteen) [inline jdaviz/configs/specviz/plugins/parsers.py:56]:
```suggestion
                data = SpectrumList.read(str(path), format=format)
        elif path.is_dir():
            data = SpectrumList.read(str(path), format=format)
```
Same as above

**Review by rosteen — CHANGES_REQUESTED** [review]
> 1 - "If someone provides a spectrum list but scalar data label, how is it supposed to work?": I don't know. The labels list must be paired with the files list. Is adding a running counter to scalar the label OK? Or should we forbid a labels list that is not paired?

As I said above, this is already handled later in the code (lines 60-68) where it checks to see if the data is a SpectrumList and whether the labels are a list or tuple. Nothing needs to be added for that in this PR.

File: jdaviz/configs/specviz/plugins/parsers.py (hunk 2 above)
Comment (ibusko) [reply jdaviz/configs/specviz/plugins/parsers.py]: Fixed

File: jdaviz/configs/specviz/plugins/parsers.py (hunk 3 above)
Comment (ibusko) [reply jdaviz/configs/specviz/plugins/parsers.py]: Fixed

**Pull request closed without merging**